payroll/search.py

```diff
@@ -75,6 +73,47 @@ def _format_results(cls, results):
         return results
 
 
+class LazyPaginatedResults(collections.abc.Sequence):
+    def __init__(self, results, hits):
+        '''
+        :results - a list of search results, `pagesize` in length
+        :hits - an int representing the total number of results
+        '''
+        self.results = results
+        self._hits = hits
+
+    def __getitem__(self, key):
+        '''
+        Django pagination accesses results via slice. For instance, if you have
+        pages of 25 items, it loads `slice(0, 25, None)` for the first page,
+        `slice(25, 50, None)` for the second page, and so on.
+
+        Because there are only `pagesize` results, we need to mock the slice
+        interface for compatibility with Django pagination. Return all results
+        if a slice is requested.
+
+        If you require sub-slices, cast LazyPaginatedResults to a list first.
+        For example:
+
+            lpr = LazyPaginatedResults(['some', 'result', 'array'], 10)
+            slice = lpr[:2]  # ['some', 'result', 'array']
+            subslice = list(lpr)[:2]  # ['some', 'result']
+        '''
+        if isinstance(key, slice):
+            return self.results
+
+        return self.results[key]
```

Comment:
Looking good.

Let's raise a not implemented error is it's not a slice. I can't really think of what the correct behavior for this should be.

https://docs.python.org/3.6/library/exceptions.html#NotImplementedError